File: src/com/l2kt/gameserver/handler/UserCommandHandler.java
package com.l2kt.gameserver.handler;

import java.util.HashMap;
import java.util.Map;

import com.l2kt.gameserver.handler.usercommandhandlers.ChannelDelete;
import com.l2kt.gameserver.handler.usercommandhandlers.ChannelLeave;
import com.l2kt.gameserver.handler.usercommandhandlers.ChannelListUpdate;
import com.l2kt.gameserver.handler.usercommandhandlers.ClanPenalty;
import com.l2kt.gameserver.handler.usercommandhandlers.ClanWarsList;
import com.l2kt.gameserver.handler.usercommandhandlers.DisMount;
import com.l2kt.gameserver.handler.usercommandhandlers.Escape;
import com.l2kt.gameserver.handler.usercommandhandlers.Loc;
import com.l2kt.gameserver.handler.usercommandhandlers.Mount;
import com.l2kt.gameserver.handler.usercommandhandlers.OlympiadStat;
import com.l2kt.gameserver.handler.usercommandhandlers.PartyInfo;
import com.l2kt.gameserver.handler.usercommandhandlers.SiegeStatus;
import com.l2kt.gameserver.handler.usercommandhandlers.Time;

public class UserCommandHandler
{
	private final Map<Integer, IUserCommandHandler> _entries = new HashMap<>();
	
	protected UserCommandHandler()
	{
		registerHandler(new ChannelDelete());
		registerHandler(new ChannelLeave());
		registerHandler(new ChannelListUpdate());
		registerHandler(new ClanPenalty());
		registerHandler(new ClanWarsList());
		registerHandler(new DisMount());
		registerHandler(new Escape());
		registerHandler(new Loc());
		registerHandler(new Mount());
		registerHandler(new OlympiadStat());
		registerHandler(new PartyInfo());
		registerHandler(new SiegeStatus());
		registerHandler(new Time());
	}
	
	private void registerHandler(IUserCommandHandler handler)
	{
		for (int id : handler.getUserCommandList())
			_entries.put(id, handler);
	}
	
	public IUserCommandHandler getHandler(int userCommand)
	{
		return _entries.get(userCommand);
	}
	
	public int size()
	{
		return _entries.size();
	}
	
	public static UserCommandHandler getInstance()
	{
		return SingletonHolder.INSTANCE;
	}
	
	private static class SingletonHolder
	{
		protected static final UserCommandHandler INSTANCE = new UserCommandHandler();
	}
}

File: src/com/l2kt/gameserver/handler/UserCommandHandler.kt
package com.l2kt.gameserver.handler

import com.l2kt.gameserver.handler.usercommandhandlers.*
import java.util.*

object UserCommandHandler {
    private val _entries = HashMap<Int, IUserCommandHandler>()

    init {
        registerHandler(ChannelDelete())
        registerHandler(ChannelLeave())
        registerHandler(ChannelListUpdate())
        registerHandler(ClanPenalty())
        registerHandler(ClanWarsList())
        registerHandler(DisMount())
        registerHandler(Escape())
        registerHandler(Loc())
        registerHandler(Mount())
        registerHandler(OlympiadStat())
        registerHandler(PartyInfo())
        registerHandler(SiegeStatus())
        registerHandler(Time())
    }

    private fun registerHandler(handler: IUserCommandHandler) {
        for (id in handler.userCommandList)
            _entries[id] = handler
    }

    fun getHandler(userCommand: Int): IUserCommandHandler? {
        return _entries[userCommand]
    }

    fun size(): Int {
        return _entries.size
    }
}